### PR TITLE
[idrac_support_assist]: run and export operation

### PR DESCRIPTION
**Pull request by ShivamSh3**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: plugins/modules/idrac_support_assist.py [hunk 1]
```diff
@@ -0,0 +1,712 @@
...
+        description:
+          - Username of the network share.
+          - I(username) is required when I(share_type) is C(cifs).
+        type: str
+      password:
+        description:
+          - Password of the network share.
+          - I(password) is required when I(share_type) is C(cifs).
+        type: str
+      ignore_certificate_warning:
```
Comment (trisha-dell) [inline plugins/modules/idrac_support_assist.py:116]: can this parameter be boolean?

File: plugins/modules/idrac_support_assist.py [hunk 2]
```diff
@@ -0,0 +1,712 @@
...
+            "elements": 'str',
+            "choices": ['hardware_data', 'storage_logs', 'os_app_data', 'debug_logs', 'telemetry_reports', 'gpu_logs']
+        },
+        "job_wait": {"type": 'bool', "default": True},
+        "job_wait_timeout": {"type": 'int', "default": 3600},
+        "share_parameters": {
+            "type": 'dict',
+            "options": {
+                "share_type": {
+                    "type": 'str',
```
Comment (trisha-dell) [inline plugins/modules/idrac_support_assist.py:907]: if the type is str, you need not mention it explicitly here.

**Review by trisha-dell — COMMENTED** [review]
What is the current UT code coverage?

File: plugins/modules/idrac_support_assist.py (hunk 1 above)
Comment (ShivamSh3) [reply plugins/modules/idrac_support_assist.py:116]: As per the iDRAC GUI and API, it has got these choices. Same was updated in the specs as well.

File: plugins/modules/idrac_support_assist.py (hunk 2 above)
Comment (ShivamSh3) [reply plugins/modules/idrac_support_assist.py:907]: It's more readable that's why we have kept like this in all the modules.

**Comment by ShivamSh3** [comment]
> What is the current UT code coverage?

image

**Comment by ShivamSh3** [comment]
image

File: tests/unit/plugins/modules/test_idrac_support_assist.py [hunk 3]
```diff
@@ -0,0 +1,1377 @@
...
+
+    def test_eula_status(self, idrac_default_args, idrac_connection_support_assist_mock, mocker):
+        obj = MagicMock()
+        mocker.patch(MODULE_PATH + API_INVOKE_MOCKER, return_value=obj)
+        f_module = self.get_module_mock(
+            params=idrac_default_args, check_mode=False)
+        eula_status_obj = self.module.AcceptEULA(
+            idrac_connection_support_assist_mock, f_module)
+        status = eula_status_obj.eula_status()
+        assert status
```
Comment (ABHISHEK-SINHA10) [inline tests/unit/plugins/modules/test_idrac_support_assist.py:398]: assert like 

status is True or False

Comment (ShivamSh3) [reply tests/unit/plugins/modules/test_idrac_support_assist.py:398]: it returns magicmock id, can't assert that.

**Comment by ShivamSh3** [comment]
image

**Review by ABHISHEK-SINHA10 — APPROVED** [review]

**Review by singh-lovepreet1 — APPROVED** [review]

**Review by rajshekarp87 — APPROVED** [review]

**Pull request merged**